src/EditorFeatures/CSharp/InlineRename/CSharpEditorInlineRenameService.cs

```diff
@@ -18,4 +26,105 @@ internal sealed class CSharpEditorInlineRenameService(
+        var references = ArrayBuilder<string>.GetInstance();
+        var docComments = ArrayBuilder<string>.GetInstance();
+
+        foreach (var renameDefinition in inlineRenameInfo.DefinitionLocations)
+        {
+            var containingStatementOrDeclarationSpan =
+                await renameDefinition.Document.TryGetSurroundingNodeSpanAsync<StatementSyntax>(renameDefinition.SourceSpan, cancellationToken).ConfigureAwait(false) ??
+                await renameDefinition.Document.TryGetSurroundingNodeSpanAsync<MemberDeclarationSyntax>(renameDefinition.SourceSpan, cancellationToken).ConfigureAwait(false);
+
+            var documentText = await renameDefinition.Document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+            if (documentText is null)
+            {
+                continue;
+            }
+
+            var symbolService = renameDefinition.Document.GetLanguageService<IGoToDefinitionSymbolService>();
+            if (symbolService is not null)
+            {
+                var textSpan = inlineRenameInfo.TriggerSpan;
+                var (symbol, _, _) = await symbolService.GetSymbolProjectAndBoundSpanAsync(
+                    renameDefinition.Document, textSpan.Start, cancellationToken)
+                    .ConfigureAwait(true);
+                var docComment = symbol?.GetDocumentationCommentXml(expandIncludes: true, cancellationToken: cancellationToken);
+                if (!string.IsNullOrWhiteSpace(docComment))
+                {
+                    docComments.Add(docComment!);
+                }
+            }
+
+            AddSpanOfInterest(documentText, renameDefinition.SourceSpan, containingStatementOrDeclarationSpan, definitions);
+        }
+
+        foreach (var renameLocation in inlineRenameLocationSet.Locations)
+        {
+            var containingStatementOrDeclarationSpan =
+                await renameLocation.Document.TryGetSurroundingNodeSpanAsync<BaseMethodDeclarationSyntax>(renameLocation.TextSpan, cancellationToken).ConfigureAwait(false) ??
+                await renameLocation.Document.TryGetSurroundingNodeSpanAsync<StatementSyntax>(renameLocation.TextSpan, cancellationToken).ConfigureAwait(false) ??
+                await renameLocation.Document.TryGetSurroundingNodeSpanAsync<MemberDeclarationSyntax>(renameLocation.TextSpan, cancellationToken).ConfigureAwait(false);
+
+            var documentText = await renameLocation.Document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+            if (documentText is null)
+            {
+                continue;
+            }
+
+            AddSpanOfInterest(documentText, renameLocation.TextSpan, containingStatementOrDeclarationSpan, references);
+        }
+
+        var context = ImmutableDictionary<string, ImmutableArray<string>>.Empty;
+        if (!definitions.IsEmpty)
+        {
+            context = context.Add("definition", definitions.ToImmutableAndFree());
```

Comment:
```suggestion
            context = context.Add("definition", definitions.ToImmutable());
```

same for below.

Reply:
create a dictionary builder instead and add to that.  this form forks the dictionary each time.